### DISCUSSION TITLE
Difficulty with updating at-spi2-atk and at-spi-core formulae.

### DISCUSSION THREAD

**Discussion by kevdogg** [Homebrew/discussions#2098] [Casks] [answered]
I wish I could provide some more insight into this error, although I've been receiving it for a while now when trying to upgrade these packages -- at-spi2-atk and at-spi-core.  I'm getting this:

```
==> Upgrading 2 outdated packages:
at-spi2-atk 2.30.1 -> 2.34.2
at-spi2-core 2.30.1_1 -> 2.36.1_3
==> Upgrading at-spi2-atk
  2.30.1 -> 2.34.2

at-spi2-atk: Linux is required for this software.
at-spi2-core: Linux is required for this software.
Error: at-spi2-atk: Unsatisfied requirements failed this build.
```

Is there a workaround for this?

**Comment by alebcay** [comment]
Are you running macOS or Linux? As mentioned in the error message, the two formulae in question are available on Linux only. They interface with D-Bus, which is not available on macOS.

If you're running on Linux and seeing this message, we would appreciate having the output of `brew doctor`, `brew --config`, and `brew --env` to help troubleshoot further.

**Reply by kevdogg** [reply]
Oh -- I honestly didn't even knew that brew ran on linux.  No I'm running on macOS.  Not sure how these packages were installed on MacOS

**Comment by cho-m** [comment, marked as answer]
These formulae were previously available on macOS and were removed back in 2019 via https://github.com/Homebrew/homebrew-core/pull/37914 as they weren't really used anymore for macOS.

If I were to guess, you may have gone through following scenario:
1. installed a formula that originally depended on `at-spi2-atk`/`at-spi2-core`
2. when these were dropped as dependencies, they were left behind on your machine as unused formulae.
3. with recent work on merging Linuxbrew into Homebrew, `at-spi2-atk`/`at-spi2-core` were re-added to Homebrew/core, but only for Linux https://github.com/Homebrew/homebrew-core/pull/78816.
4. Homebrew sees that you have `at-spi2-atk`/`at-spi2-core` installed and tries to update to "newer" Linux-only release, which causes failure.

If you aren't directly using these, you should uninstall them.

**Reply by kevdogg** [reply]
Thanks - I just went ahead and manually removed those two packages.
